### PR TITLE
#1758 Whitespace after line continuation char causes it to be ignored

### LINKED ISSUE #1758: Whitespace after line continuation character causes it to be ignored
`\` as the final char on a line works: 

```
ksql> SHOW \
> STREAMS;

 Stream Name | Kafka Topic | Format
------------------------------------
 SHIPMENTS   | shipments   | JSON
------------------------------------
```

However if some whitespace (e.g. a space or tab) exists after the `\`, the continuation is not honoured and KSQL throws an error because it just parses the input on that line. 

This is something that will commonly occur in copying and pasting snippets from tutorials, emails, etc etc. 

See https://asciinema.org/a/HMx5H3d4KaVBRYeYgHtHG4g7c for an illustration of this. 

[asciicast](https://asciinema.org/a/HMx5H3d4KaVBRYeYgHtHG4g7c)

Related: #1425

### PR DESCRIPTION
**Pull request by uurl**
### Description 
This PR fixes #1758 Whitespace after line continuation char causes it to be ignored

### Testing done 
mvn clean package

### Reviewer checklist
- [x] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [x] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Comment by uurl** [comment]
@big-andy-coates @apurvam @rodesai review please

File: ksql-cli/src/main/java/io/confluent/ksql/cli/Cli.java [hunk 1]
```diff
@@ -213,16 +213,17 @@ public void runNonInteractively(final String input) throws Exception {
...
     final List<String> result = new ArrayList<>();
     StringBuilder logicalLine = new StringBuilder();
     for (final String physicalLine : input.split("\n")) {
-      if (!physicalLine.trim().isEmpty()) {
-        if (physicalLine.endsWith("\\")) {
-          logicalLine.append(physicalLine.substring(0, physicalLine.length() - 1));
+      final String trimmedPhysicalLine = physicalLine.trim();
+      if (!trimmedPhysicalLine.isEmpty()) {
+        if (trimmedPhysicalLine.endsWith("\\")) {
+          logicalLine.append(trimmedPhysicalLine.replaceAll("\\\\", " "));
```
Comment (big-andy-coates) [inline ksql-cli/src/main/java/io/confluent/ksql/cli/Cli.java]: I don't think we want to replace _all_ slashes, only the last slash. I think the only change you require it to `trim()` the `physicalLine`, as you have, and use it everywhere `physicalLine` was used.

Comment (big-andy-coates) [inline ksql-cli/src/main/java/io/confluent/ksql/cli/Cli.java]: I don't think we want to replace all slashes, only the last slash. I think the only change you require it to trim() the physicalLine, as you have, and use it everywhere physicalLine was used.

File: ksql-cli/src/main/java/io/confluent/ksql/cli/Cli.java [hunk 2]
```diff
@@ -213,16 +213,17 @@ public void runNonInteractively(final String input) throws Exception {
...
         } else {
-          result.add(logicalLine.append(physicalLine).toString().trim());
+          result.add(logicalLine.append(trimmedPhysicalLine.replaceAll("\\\\", " ")).toString());
```
Comment (big-andy-coates) [inline ksql-cli/src/main/java/io/confluent/ksql/cli/Cli.java]: There should be no need to replace slashes in this line as it does not end with a slash.

File: ksql-cli/src/main/java/io/confluent/ksql/cli/Cli.java (hunk 1 above)
Comment (uurl) [reply ksql-cli/src/main/java/io/confluent/ksql/cli/Cli.java]: The same question as in #1425.
I guess that all these commands should run and return the same result:

`SHOW TABLES;`

`SHOW\        `
`TABLES;`

`SHOW\     TABLES;`

`SHOW\TABLES;`

**Comment by big-andy-coates** [comment]
Closing, as superseded by #1846.

Thanks for all your recent efforts @uurl.

**Pull request closed without merging**